Sources/Purchasing/Purchases/PurchasesType.swift

```diff
@@ -0,0 +1,943 @@
+
+    /**
+     * This function will log in the current user with an ``appUserID``.
+     *
+     * - Parameter appUserID: The ``appUserID`` that should be linked to the current user.
+     * - returns: A tuple of: the latest ``CustomerInfo`` and a `Bool` specifying
+     * whether the user was created for the first time in the RevenueCat backend.
+     *
+     * RevenueCat provides a source of truth for a subscriber's status across different platforms.
+     * To do this, each subscriber has an App User ID that uniquely identifies them within your application.
+     *
+     * User identity is one of the most important components of many mobile applications,
+     * and it's extra important to make sure the subscription status RevenueCat is
+     * tracking gets associated with the correct user.
+     *
+     * The Purchases SDK allows you to specify your own user identifiers or use anonymous identifiers
+     * generated by RevenueCat. Some apps will use a combination
+     * of their own identifiers and RevenueCat anonymous Ids - that's okay!
+     *
+     * #### Related Articles
+     * - [Identifying Users](https://docs.revenuecat.com/docs/user-ids)
+     * - ``logOut()``
+     * - ``isAnonymous``
+     * - ``Purchases/appUserID``
+     */
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    func logIn(_ appUserID: String) async throws -> (customerInfo: CustomerInfo, created: Bool)
+
+    /**
+     * Logs out the ``Purchases`` client, clearing the saved ``appUserID``.
+     *
+     * This will generate a random user id and save it in the cache.
+     * If this method is called and the current user is anonymous, it will return an error.
+     *
+     * #### Related Articles
+     * - [Identifying Users](https://docs.revenuecat.com/docs/user-ids)
+     * - ``logIn(_:completion:)``
+     * - ``isAnonymous``
+     * - ``Purchases/appUserID``
+     */
+    func logOut(completion: ((CustomerInfo?, PublicError?) -> Void)?)
+
+    /**
+     * Logs out the ``Purchases`` client, clearing the saved ``appUserID``.
+     *
+     * This will generate a random user id and save it in the cache.
+     * If this method is called and the current user is anonymous, it will return an error.
+     *
+     * #### Related Articles
+     * - [Identifying Users](https://docs.revenuecat.com/docs/user-ids)
+     * - ``logIn(_:)``
+     * - ``isAnonymous``
+     * - ``Purchases/appUserID``
+     */
+    @available(iOS 13.0, macOS 10.15, tvOS 13.0, watchOS 6.2, *)
+    func logOut() async throws -> CustomerInfo
+
+    /**
+     * Get latest available customer  info.
+     *
+     * - Parameter completion: A completion block called when customer info is available and not stale.
+     * Called immediately if ``CustomerInfo`` is cached. Customer info can be nil if an error occurred.
+     */
+    func getCustomerInfo(completion: @escaping ((CustomerInfo?, PublicError?) -> Void))
+
+    /**
+     * Get latest available customer  info.
```

Comment:
Not sure if this was before but double space 
```suggestion
     * Get latest available customer info.
```